pose_broadcaster/test/test_pose_broadcaster.hpp

```diff
@@ -77,7 +78,10 @@ void PoseBroadcasterTest::subscribe_and_get_message(const std::string & topic, T
   constexpr size_t max_sub_check_loop_count = 5;
   for (size_t i = 0; !received_msg; ++i)
   {
-    ASSERT_LT(i, max_sub_check_loop_count);
+    if (i >= max_sub_check_loop_count)
+    {
+      throw std::runtime_error("Failed to receive message on topic: " + topic);
+    }
```

Comment:
```suggestion
    ASSERT_LT(i, max_sub_check_loop_count) << "Failed to receive message on topic: " << topic;
```

Will this work?

Is there any specific reason for throwing?

Reply:
Yes, because this is expected to fail on an illegal quaternion. Throwing an exception here allows using an `EXPECT_THROW` statement around this function from the test.

Reply:
Got it